seatunnel-engine/seatunnel-engine-client/src/main/java/org/apache/seatunnel/engine/client/SeaTunnelHazelcastClient.java

```diff
@@ -0,0 +1,106 @@
+
+import com.hazelcast.client.impl.clientside.HazelcastClientInstanceImpl;
+import com.hazelcast.client.impl.clientside.HazelcastClientProxy;
+import com.hazelcast.client.impl.protocol.ClientMessage;
+import com.hazelcast.client.impl.spi.impl.ClientInvocation;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.serialization.SerializationService;
+import com.hazelcast.internal.util.Preconditions;
+import com.hazelcast.logging.ILogger;
+import lombok.NonNull;
+
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+
+public class SeaTunnelHazelcastClient {
+    private final HazelcastClientInstanceImpl hazelcastClient;
+    private final SerializationService serializationService;
+
+    public SeaTunnelHazelcastClient(@NonNull SeaTunnelClientConfig seaTunnelClientConfig) {
+        Preconditions.checkNotNull(seaTunnelClientConfig, "config");
+        this.hazelcastClient =
+            ((HazelcastClientProxy) com.hazelcast.client.HazelcastClient.newHazelcastClient(
+                seaTunnelClientConfig)).client;
+        this.serializationService = hazelcastClient.getSerializationService();
+        ExceptionUtil.registerSeaTunnelExceptions(hazelcastClient.getClientExceptionFactory());
+    }
+
+    public SerializationService getSerializationService() {
+        return serializationService;
+    }
+
+    /**
+     * Returns the underlying Hazelcast IMDG instance used by SeaTunnel Engine Client. It will
+     * be a client, depending on the type of this
+     */
+    @NonNull
+    public HazelcastInstance getHazelcastInstance() {
+        return hazelcastClient;
+    }
+
+    public ILogger getLogger(Class<?> clazz) {
+        return hazelcastClient.getLoggingService().getLogger(clazz);
+    }
+
+    public <S> S requestOnMasterAndDecodeResponse(@NonNull ClientMessage request,
+                                                  @NonNull Function<ClientMessage, Object> decoder) {
+        UUID masterUuid = hazelcastClient.getClientClusterService().getMasterMember().getUuid();
+        return requestAndDecodeResponse(masterUuid, request, decoder);
+    }
+
+    public <S> S requestOnAnyMemberAndDecodeResponse(@NonNull ClientMessage request,
+                                                     @NonNull Function<ClientMessage, Object> decoder) {
+        return requestAndDecodeResponse(null, request, decoder);
+    }
+
+    public <S> S requestAndDecodeResponse(UUID uuid, ClientMessage request,
+                                          Function<ClientMessage, Object> decoder) {
+        ClientInvocation invocation = new ClientInvocation(hazelcastClient, request, null, uuid);
+        try {
+            ClientMessage response = invocation.invoke().get();
+            return serializationService.toObject(decoder.apply(response));
+        } catch (Throwable t) {
+            throw ExceptionUtil.rethrow(t);
+        }
```

Comment:
invocation.invoke().get() may throw an InterruptedException.
we don't ignore this.you need restore interrupted state

Reply:
done